util/dnsname/dnsname_test.go

```diff
@@ -230,3 +231,54 @@ func BenchmarkToFQDN(b *testing.B) {
 		})
 	}
 }
+
+func TestValidLabel(t *testing.T) {
```

Comment:
Please add a test case for `_label` forms as well. That's what I broke the last time I tried to make dnsname more strict :)

Reply:
`dnsname.ValidLabel` already rejects that, this is for generating hostnames. But yeah, that's a big mess.